### PR TITLE
Solflare signAndSendTransaction

### PR DESCRIPTION
**Pull request by vsakos**
Implemented `sendTransaction` in the Solflare adapter using the `signAndSendTransaction` extension method.

### CODE REVIEW

File: packages/wallets/solflare/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@solana/wallet-adapter-solflare",
-    "version": "0.6.24",
+    "version": "0.6.25",
```
Comment (jordansexton) [inline packages/wallets/solflare/package.json:3]: Please revert this change, the changeset will handle version updates. :)

File: packages/wallets/solflare/src/adapter.ts [hunk 2]
```diff
@@ -2,21 +2,25 @@ import type { WalletAdapterNetwork, WalletName } from '@solana/wallet-adapter-ba
...
     WalletNotReadyError,
     WalletPublicKeyError,
     WalletReadyState,
+    WalletSendTransactionError,
     WalletSignMessageError,
     WalletSignTransactionError,
 } from '@solana/wallet-adapter-base';
 import type { Transaction, TransactionVersion, VersionedTransaction } from '@solana/web3.js';
-import { PublicKey } from '@solana/web3.js';
+import { Connection, PublicKey, TransactionSignature } from '@solana/web3.js';
```
Comment (jordansexton) [inline packages/wallets/solflare/src/adapter.ts:23]:
```suggestion
import { type Connection, PublicKey, type TransactionSignature } from '@solana/web3.js';
```

Must be imported as types since they are only used as types. Run `pnpm run lint` to check.

File: packages/wallets/solflare/src/adapter.ts [hunk 3]
```diff
@@ -2,21 +2,25 @@ import type { WalletAdapterNetwork, WalletName } from '@solana/wallet-adapter-ba
 import {
     BaseMessageSignerWalletAdapter,
     isIosAndRedirectable,
+    isVersionedTransaction,
     scopePollingDetectionStrategy,
+    SendTransactionOptions,
```
Comment (jordansexton) [inline packages/wallets/solflare/src/adapter.ts:7]:
```suggestion
    type SendTransactionOptions,
```

Must be imported as types since they are only used as types. Run `pnpm run lint` to check.

File: packages/wallets/solflare/src/adapter.ts [hunk 4]
```diff
@@ -175,6 +179,39 @@ export class SolflareWalletAdapter extends BaseMessageSignerWalletAdapter {
...
+                if (isVersionedTransaction(transaction)) {
+                    signers?.length && transaction.sign(signers);
+                } else {
+                    transaction = (await this.prepareTransaction(transaction, connection, sendOptions)) as T;
+                    signers?.length && (transaction as Transaction).partialSign(...signers);
+                }
+
+                sendOptions.preflightCommitment = sendOptions.preflightCommitment || connection.commitment;
+
+                return await wallet.signAndSendTransaction(transaction, sendOptions);
```
Comment (jordansexton) [inline packages/wallets/solflare/src/adapter.ts:203]: `signAndSendTransaction` is not a declared method on the exported types of the Solflare class. The SDK will need to be updated to fix this.

File: packages/wallets/solflare/src/adapter.ts [hunk 5]
```diff
@@ -175,6 +179,39 @@ export class SolflareWalletAdapter extends BaseMessageSignerWalletAdapter {
...
+            } catch (error: any) {
+                console.log(error);
```
Comment (jordansexton) [inline packages/wallets/solflare/src/adapter.ts:205]:
```suggestion
```

Please remove logging.

File: packages/wallets/solflare/package.json [hunk 6]
```diff
@@ -36,7 +36,7 @@
     },
     "dependencies": {
         "@solana/wallet-adapter-base": "workspace:^",
-        "@solflare-wallet/sdk": "^1.2.1"
+        "@solflare-wallet/sdk": "^1.3.0"
```
Comment (jordansexton) [inline packages/wallets/solflare/package.json:39]: Will need to be updated with a type fix, see below.

**Review by jordansexton — CHANGES_REQUESTED** [review]
Thanks! Needs a few fixes. Please run `pnpm lint:fix` and `pnpm build` to check for issues.

**Comment by jordansexton** [comment]
I committed the other fixes, but the type fix to the SDK will need to be made upstream first.

File: packages/wallets/solflare/package.json (hunk 6 above)
Comment (vsakos) [reply packages/wallets/solflare/package.json:39]: @jordansexton Can you tell me which type is missing? The source is in TS so the declarations are generated automatically and we have this in both `lib/cjs/index.d.ts` and `lib/esm/index.d.ts`:

```
signAndSendTransaction(transaction: TransactionOrVersionedTransaction, options?: SendOptions): Promise<string>;
```

Comment (jordansexton) [reply packages/wallets/solflare/package.json:39]: Hmm, you must be right, I went back into the project branch and the method was there. Must have been a package installation problem.
